packages/next/src/server/request/params.browser.ts

```diff
@@ -0,0 +1,174 @@
+import type { Params } from './params'
+
+import { ReflectAdapter } from '../web/spec-extension/adapters/reflect'
+import { InvariantError } from '../../shared/lib/invariant-error'
+
+export function reifyClientRenderParams(underlying: Params) {
+  if (process.env.NODE_ENV === 'development') {
+    return makeDynamicallyTrackedExoticParamsWithDevWarnings(underlying)
+  } else {
+    return makeUntrackedExoticParams(underlying)
+  }
+}
+
+interface CacheLifetime {}
+const CachedParams = new WeakMap<CacheLifetime, Promise<Params>>()
+
+function makeUntrackedExoticParams(underlying: Params): Promise<Params> {
+  const cachedParams = CachedParams.get(underlying)
+  if (cachedParams) {
+    return cachedParams
+  }
+
+  const promise = Promise.resolve(underlying)
+  CachedParams.set(underlying, promise)
+
+  Object.defineProperties(promise, {
+    status: {
+      value: 'fulfilled',
+      writable: true,
+    },
+    value: {
+      value: underlying,
+      writable: true,
+    },
+  })
+
+  Object.keys(underlying).forEach((prop) => {
+    switch (prop) {
+      case 'then':
+      case 'value':
+      case 'status': {
+        // These properties cannot be shadowed with a search param because they
+        // are necessary for ReactPromise's to work correctly with `use`
```

Comment:
are we warning about this anywhere? `value` and `status` seem like not-unlikely search param names

Reply:
I feel we coud give a warning here, as the access to `value` and `status` can be everywhere, would be tricky to let codemod to catch them.